Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM debian:testing
+
+# Install dependencies
+RUN \
+  apt-get update && \
+  apt-get install -y git python3-pip python3-venv && \
+  pip3 install keyring && \
+  git clone git://github.com/psf/requests.git && \
+  cd requests && \
+  pip3 install .
+
+# Install bitbucket-issue-migration
+RUN git clone https://github.com/jeffwidman/bitbucket-issue-migration.git && \
+  cd bitbucket-issue-migration && \
+  pip3 install -r requirements.pip
```

Comment:
Why don't we instead make this the first thing after installing python3?

That way we don't have to track the dependencies two places...

I'm fine with changing `keyring` to be a specified / non-optional dependency in `requirements.pip`...

Reply:
One (small) benefit of keeping the dependencies as a separate step is that step can be cached even if the local repo changes. I'm using this to run locally so it was helpful for me to iterate, but it probably isn't that important once submitted.

I'd vote to keep keyring an optional dep (it is optional, after all), but I'm not sure the best way to express that here without introducing some redundancy. I poked around with [extras](https://www.python.org/dev/peps/pep-0508/#extras), i.e. `keyring; extra == 'key'`, which works in requirements.pip but I can't figure out how to use `-r` and also specify `keyring` as an extra. I get a `Ignoring keyring: markers 'extra == "key"' don't match your environment` message from `pip`. I _think_ [using `setup.py` would work](https://caremad.io/posts/2013/07/setup-vs-requirement/), but I don't really want to mess with that here.

My suggestion would be to keep keyring optional and tolerate the redundancy for now; it is just one dep after all. If things get more complex a better optional-dep-management strategy would then be worth investigating further. I've updated the PR, LMKWYT of this approach.

Reply:
This looks reasonable for how you implement the installation.